ui/src/components/NewWorkflow/HTTPTask.tsx

```diff
@@ -0,0 +1,205 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import { Form, Formik } from 'formik'
+import { FormControlLabel, Switch } from '@material-ui/core'
+import { MenuItem, Typography } from '@material-ui/core'
+import { SelectField, Submit, TextField } from 'components/FormField'
+import { Template, TemplateType } from 'slices/workflows'
+import { parseHTTPTask, renderHTTPTask } from 'api/workflows'
+import { useEffect, useRef, useState } from 'react'
+
+import Paper from 'components-mui/Paper'
+import PaperTop from 'components-mui/PaperTop'
+import { RequestForm } from 'api/workflows.type'
+import Space from 'components-mui/Space'
+import T from 'components/T'
+import { makeStyles } from '@material-ui/styles'
+import { useIntl } from 'react-intl'
+import { validateName } from 'lib/formikhelpers'
+
+const useStyles = makeStyles({
+  field: {
+    width: 180,
+  },
+})
+
+const HTTPMethods = ['GET', 'POST', 'PUT', 'DELETE', 'PATCH', 'HEAD', 'OPTION']
+
+interface HTTPTaskProps extends CommonTemplateProps {
+  childrenCount: number
+  submitTemplate: (template: Template) => void
+}
+
+interface CommonTemplateProps {
+  name?: string
+  deadline?: string
+  type: TemplateType
+  templates: Template[]
+  externalTemplate?: Template
+}
+
+const HTTPTask: React.FC<HTTPTaskProps> = (props) => {
+  const intl = useIntl()
+  const classes = useStyles()
+
+  const { submitTemplate } = props
+  const onSubmit = (form: RequestForm) => {
+    renderHTTPTask(form)
+      .then((response) => {
+        const { name, task } = response.data!
+        const { container } = task
+        const result: Template = {
+          name,
+          children: [],
+          type: TemplateType.Custom,
+          custom: {
+            container,
+            conditionalBranches: [],
+          },
+        }
+        submitTemplate(result)
+      })
+      .catch(console.error)
+  }
+  const formRef = useRef<any>()
+  const [initialValues, setInitialValues] = useState<RequestForm>({
+    name: props.name || '',
+    url: '',
+    method: '',
+    body: '',
+    followLocation: false,
+    jsonContent: false,
+  })
+
+  const validateRequestForm = (newValue: RequestForm) => {
+    console.log(newValue)
+    const errors: any = {}
+    return errors
+  }
+
+  useEffect(() => {
+    if (props.externalTemplate) {
+      // TODO: use unified name
+      const backendType = props.externalTemplate.type === 'custom' ? 'Task' : props.externalTemplate.type
+      parseHTTPTask({
+        name: props.externalTemplate.name,
+        templateType: backendType,
+        task: {
+          container: props.externalTemplate.custom!.container,
+          conditionalBranches: props.externalTemplate.custom!.conditionalBranches,
+        },
+      })
+        .then((response) => {
+          if (response.data) {
+            const parsedForm = response.data as RequestForm
+            setInitialValues({
```

Comment:
```js
{
  ...parsedForm,
  followLocation: parsedForm.followLocation || false,
  jsonContent: parsedForm.jsonContent || false,
}
```

I think this block can be written like above if the fields are exact.

Reply:
done in c86efddefbce0c03476e207bb024b9d395e9b4c1